taletime/lib/share/screens/shared_story.dart

```diff
@@ -0,0 +1,244 @@
+  String? _storyId;
+  DocumentReference<Story>? _storyRef;
+
+  @override
+  void initState() {
+    super.initState();
+
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      setState(() {
+        _storyId = GoRouterState.of(context).uri.queryParameters["storyId"];
+
+        _storyRef = FirebaseFirestore.instance
+            .doc("allStories/$_storyId")
+            .withConverter(
+              fromFirestore: (snap, _) => Story.fromDocumentSnapshot(snap),
+              toFirestore: (snap, _) => snap.toFirebase(),
+            );
+      });
+    });
+  }
+
+  Widget _buildStoryMetadata(BuildContext context, Story? story) {
+    final double screenWidth = MediaQuery.of(context).size.width;
+    final double screenHeight = MediaQuery.of(context).size.height;
+    final imageSize = min(screenHeight, screenWidth) * 0.6;
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.center,
+      children: [
+        Container(
+          width: imageSize,
+          height: imageSize,
+          decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(15),
+            color: Colors.transparent,
+          ),
+          child: Image(
+              fit: BoxFit.cover,
+              image: story?.imageUrl != null
+                  ? NetworkImage(story!.imageUrl!)
+                  : const AssetImage("assets/logo.png")
+                      as ImageProvider<Object>),
+        ),
+        Padding(
+          padding: const EdgeInsets.all(12.0),
+          child: Text(
+            story?.title ?? AppLocalizations.of(context)!.noTitle,
+            softWrap: true,
+            style: const TextStyle(
+                color: Colors.teal,
+                fontSize: 21.0,
+                fontWeight: FontWeight.bold),
+          ),
+        ),
+        Padding(
+          padding: const EdgeInsets.all(8.0),
+          child: Text(
+            story?.author ?? AppLocalizations.of(context)!.noName,
+            softWrap: true,
+            style: const TextStyle(
+              color: Colors.teal,
+              fontSize: 16.0,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildAddToProfile(BuildContext context, Story? story) {
+    renderWidget(
+        {required Widget profile, Widget? error, void Function()? btnClick}) {
+      return Column(
+        children: [
+          Padding(
+            padding: const EdgeInsets.symmetric(vertical: 8),
+            child: profile,
+          ),
+          error ?? const SizedBox(),
+          ElevatedButton(
+            style: elevatedButtonDefaultStyle(),
+            onPressed: btnClick,
+            child: Text(AppLocalizations.of(context)!.addToProfile),
+          ),
+        ],
+      );
+    }
+
+    // Test if logged in
+    if (Provider.of<UserState>(context).user == null) {
+      return renderWidget(
+        profile: Text(AppLocalizations.of(context)!.notLoggedIn),
+        btnClick: story == null
+            ? null
```

Comment:
Don't know what is more readable. Simply story or null

Reply:
This would lead to incompatible types.